scene/gui/tab_container.cpp

```diff
@@ -479,7 +481,14 @@ void TabContainer::set_current_tab(int p_current) {
 	}
 
 	_change_notify("current_tab");
-	emit_signal("tab_changed",current);
+
+	if (pending_previous == current)
+		emit_signal("tab_selected", current);
```

Comment:
I think that's confusing, the name does not convey that it would be emitted only when the tab being selected is the current tab. Based on the names of those signals, I'd expect both `tab_selected` and `tab_changed` to be emitted when clicking on a different tab.